R/package_hooks_linter.R

```diff
@@ -91,10 +91,14 @@ package_hooks_linter <- function() {
     # NB: base only checks the SYMBOL_FUNCTION_CALL version, not SYMBOL.
     library_require_xpath <- "
     //expr[SYMBOL[text() = '.onAttach' or text() = '.onLoad']]
-    /following-sibling::expr//*[
-      (self::SYMBOL or self::SYMBOL_FUNCTION_CALL)
-      and (text() = 'require' or text() = 'library' or text() = 'installed.packages')
-    ]
+    /following-sibling::expr
```

Comment:
again different... benchmark is for //* which is "start from root and search", which is way more overhead than "search all siblings" which is almost always one or two nodes